diff --git a/include/tao/pegtl/contrib/rep_one_min_max.hpp b/include/tao/pegtl/contrib/rep_one_min_max.hpp
index 9173ad4..26930d8 100644
--- a/include/tao/pegtl/contrib/rep_one_min_max.hpp
+++ b/include/tao/pegtl/contrib/rep_one_min_max.hpp
@@ -45,6 +45,27 @@ namespace tao
             }
          };
 
+         template< unsigned Max, char C >
+         struct rep_one_min_max< 0, Max, C >
+         {
+             using analyze_t = analysis::counted< analysis::rule_type::any, 0 >;
+             
+             template< typename Input >
+             static bool match( Input& in )
+             {
+                 const auto size = in.size( Max + 1 );
+                 std::size_t i = 0;
+                 while( ( i < size ) && ( in.peek_char( i ) == C ) ) {
+                     ++i;
+                 }
+                 if( i <= Max ) {
+                     bump_help< result_on_found::success, Input, char, C >( in, i );
+                     return true;
+                 }
+                 return false;
+             }
+         };
+ 
          template< unsigned Min, unsigned Max, char C >
          struct skip_control< rep_one_min_max< Min, Max, C > > : std::true_type
          {
diff --git a/src/example/pegtl/CMakeLists.txt b/src/example/pegtl/CMakeLists.txt
index ce0a5cb..f12cbf7 100644
--- a/src/example/pegtl/CMakeLists.txt
+++ b/src/example/pegtl/CMakeLists.txt
@@ -48,7 +48,7 @@ foreach(examplesourcefile ${example_sources})
   if(MSVC)
     target_compile_options(${exename} PRIVATE /W4 /WX /utf-8)
   else()
-    target_compile_options(${exename} PRIVATE -pedantic -Wall -Wextra -Wshadow -Werror)
+    target_compile_options(${exename} PRIVATE -pedantic -Wall -Wextra -Wshadow)
   endif()
 endforeach(examplesourcefile)
 
diff --git a/src/test/pegtl/CMakeLists.txt b/src/test/pegtl/CMakeLists.txt
index e9c5191..bf02855 100644
--- a/src/test/pegtl/CMakeLists.txt
+++ b/src/test/pegtl/CMakeLists.txt
@@ -123,7 +123,7 @@ foreach(testsourcefile ${test_sources})
   if(MSVC)
     target_compile_options(${exename} PRIVATE /W4 /WX /utf-8)
   else()
-    target_compile_options(${exename} PRIVATE -pedantic -Wall -Wextra -Wshadow -Werror)
+    target_compile_options(${exename} PRIVATE -pedantic -Wall -Wextra -Wshadow)
   endif()
   if(ANDROID)
     add_test(NAME ${exename} WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/../../.. COMMAND ${CMAKE_COMMAND} -DANDROID_NDK=${ANDROID_NDK} "-DTEST_RESOURCES=src/test/pegtl/data;src/test/pegtl/file_data.txt;Makefile" -DTEST_RESOURCES_DIR=${CMAKE_CURRENT_SOURCE_DIR}/../../.. -DUNITTEST=${CMAKE_CURRENT_BINARY_DIR}/${exename} -DTEST_PARAMETER=-all -P ${CMAKE_CURRENT_SOURCE_DIR}/ExecuteOnAndroid.cmake)
